python/drned_xmnr/op/base_op.py

```diff
@@ -321,6 +320,53 @@ def find_pytest_executable(self):
                 return executable
         raise ActionError('PyTest not installed - pytest executable not found')
 
+    def get_authgroup_info(self, trans, root, locuser, authmap):
+        if authmap.same_user.exists():
+            username = locuser
+        else:
+            username = authmap.remote_name
+        if username is None:
+            return None, None
+        if authmap.same_pass.exists():
+            upwd = root.aaa.authentication.users.user[locuser].password
+        else:
+            upwd = authmap.remote_password
+        if upwd is None:
+            return username, None
+        trans.maapi.install_crypto_keys()
+        return username, _ncs.decrypt(upwd)
+
+    def get_devcli_params(self, trans):
+        root = maagic.get_root(trans)
+        device_node = root.devices.device[self.dev_name]
+        ip = device_node.address
+        port = device_node.port
+        driver = device_node.drned_xmnr.driver
+        if driver is None:
+            raise ActionError('device driver not configured, cannot continue')
+        authgroup_name = device_node.authgroup
+        authgroup = root.devices.authgroups.group[authgroup_name]
+        locuser = self.uinfo.username
+        if locuser in authgroup.umap:
+            authmap = authgroup.umap[locuser]
+        else:
+            authmap = authgroup.default_map
+        user, passwd = self.get_authgroup_info(trans, root, locuser, authmap)
+        return driver, user, passwd, ip, port
+
+    def devcli_run(self, script, script_args):
+        driver, username, passwd, ip, port = self.run_with_trans(self.get_devcli_params)
+        args = ['python', script, '--devname', self.dev_name,
```

Comment:
I think you missed the latest change, it adds `cli-port` and tries to use that - see the second commit.

Reply:
It's really strange. I submitted both these comments two weeks ago and you didn't get notified until I approved the pull-request.

I also wrote a comment for you latest commit wondering if we should have a default value (22) for the CLI port, rather than the same as the NETCONF port if not set. but that comment is also gone... ?!?

Reply:
I think that's because you started adding comments, but they are private ("pending") until you press a button titled "Finish review" or something in this vein; or until you approve the PR, as it turns out.